### PR TITLE
Fix and improvements to the demo

### LINKED ISSUE #77: As a framework user I want a working example of the framework
**Initial estimate of effort:**
15

#### Sprint tasks
- [x] #78
- [x] #86
- [x] #87
- [x] #88
- [x] #89
- [x] #90
- [x] #91
- [x] #92
- [x] #105
- [x] #106
- [x] #107

### LINKED ISSUE #105: Improve PlayState in demo

### LINKED ISSUE #106: Add parameters tuning to demo

### PR DESCRIPTION
**Pull request by nicolasfara**
In this PR fix/introduce:
- Parallel execution of the tests in sbt
- Add Finite State Machine to control the game state
- Add UI element in order to tune `Wall restitution` and `Friction coefficient`
- Refactor `PlayState`, `MouseState` and `EnvironmentState` in order to avoid race condition in test execution

Closes #105 
Closes #106 
Closes #77

### CODE REVIEW

File: demo/src/test/scala/dev/atedeg/ecscalademo/fixtures/VelocityFixture.scala [hunk 1]
```diff
@@ -1,16 +1,17 @@
 package dev.atedeg.ecscalademo.fixtures
 
-import scala.language.implicitConversions
 import dev.atedeg.ecscala.util.types.given
 import dev.atedeg.ecscala.World
 import dev.atedeg.ecscala.dsl.ECScalaDSL
-import dev.atedeg.ecscalademo.given
```
Comment (ndido98) [inline demo/src/test/scala/dev/atedeg/ecscalademo/fixtures/VelocityFixture.scala:7]: Did you intentionally remove the `given`?

**Review by ndido98 — CHANGES_REQUESTED** [review]

File: demo/src/test/scala/dev/atedeg/ecscalademo/fixtures/VelocityFixture.scala (hunk 1 above)
Comment (nicolasfara) [reply demo/src/test/scala/dev/atedeg/ecscalademo/fixtures/VelocityFixture.scala:7]: Yes, is not used in the trait

File: demo/src/test/scala/dev/atedeg/ecscalademo/systems/VelocityArrowSystemTest.scala [hunk 2]
```diff
@@ -11,30 +12,37 @@ import org.mockito.Mockito.{ times, verify }
...
+    lazy val entity1 = world hasAn entity withComponents { Position(0, 0) &: Velocity(10, 10) }
+    lazy val canvas = mock[ECSCanvas]
+    lazy val arrowSystem = new VelocityArrowSystem(playState, mouseState, canvas)
+  }
+
   "A VelocityArrowSystem" should {
-    "run if the game is paused, there is a selected ball and the game is in velocity editing mode" in new VelocityFixture {
-      val canvas = mock[ECSCanvas]
-      val arrowSystem = VelocityArrowSystem(canvas)
+    "run if the game is paused, there is a selected ball and the game is in velocity editing mode" in new VelocityArrowFixture {
```
Comment (giacomocavalieri) [inline demo/src/test/scala/dev/atedeg/ecscalademo/systems/VelocityArrowSystemTest.scala:30]:
```suggestion
    "run if the game is in changeVelocity mode and there is a selected ball" in new VelocityArrowFixture {
```

File: demo/src/test/scala/dev/atedeg/ecscalademo/systems/VelocityArrowSystemTest.scala [hunk 3]
```diff
@@ -11,30 +12,37 @@ import org.mockito.Mockito.{ times, verify }
...
+      mouseState.coordinates = (10.0, 10.0)
+
       world hasA system(arrowSystem)
-      PlayState.selectedBall = Some(entity1)
-      PlayState.playing = false
-      MouseState.coordinates = (10.0, 10.0)
+
+      arrowSystem.shouldRun shouldBe true
       world.update(10)
-      verify(canvas, times(1)).drawLine(is(Point(0, 0)), is(Point(10, 10)), any(), anyDouble())
```
Comment (giacomocavalieri) [inline demo/src/test/scala/dev/atedeg/ecscalademo/systems/VelocityArrowSystemTest.scala:37]: Why did you remove the `times(1)`?

File: demo/src/test/scala/dev/atedeg/ecscalademo/systems/VelocityEditingSystemTest.scala [hunk 4]
```diff
@@ -1,34 +1,48 @@
...
-  def testNewExpectedVelocity(mouseCoordinates: Point, expectedVelocity: Velocity): Unit = new VelocityFixture {
-    PlayState.selectedBall = Some(entity1)
-    MouseState.clicked = true
-    MouseState.coordinates = mouseCoordinates
-    world.update(10)
+  private def testNewExpectedVelocity(
+      wrld: World,
+      sys: VelocityEditingSystem,
+      pState: PlayState,
+      mState: MouseState,
```
Comment (giacomocavalieri) [inline demo/src/test/scala/dev/atedeg/ecscalademo/systems/VelocityEditingSystemTest.scala:36]: I really can't stand abbreviations
```suggestion
      world: World,
      system: VelocityEditingSystem,
      playState: PlayState,
      mouseState: MouseState,
```

**Review by giacomocavalieri — CHANGES_REQUESTED** [review]

**Review by vitlinda — APPROVED** [review]

File: demo/src/test/scala/dev/atedeg/ecscalademo/systems/VelocityArrowSystemTest.scala (hunk 3 above)
Comment (nicolasfara) [reply demo/src/test/scala/dev/atedeg/ecscalademo/systems/VelocityArrowSystemTest.scala:37]: `Verify()` implicit assume that the method is called once

File: demo/src/test/scala/dev/atedeg/ecscalademo/systems/VelocityEditingSystemTest.scala (hunk 4 above)
Comment (nicolasfara) [reply demo/src/test/scala/dev/atedeg/ecscalademo/systems/VelocityEditingSystemTest.scala:36]: Those abbreviation in needed due a clash of name in the fixture. If clash happen the test fail

**Review by ndido98 — APPROVED** [review]

**Review by giacomocavalieri — APPROVED** [review]

**Pull request merged**